### PR TITLE
Updates to (intermediate) terms

### PR DESCRIPTION
**Pull request by mortenpi**
With #64 in mind, but also other edits.

* I added `.nu` as a property to `IntermediateTerm`. I think ideally we'd provide such fallbacks in all the cases where we're using Unicode as fields.

* I am documenting the various fields as the official API to access e.g. the L & S quantum numbers of a term etc.

* Probably not worth it, but one could rename `Term -> LSTerm`, to make it more explicit. Theoretically, one could also have `LSJTerm` (or should it be [`LSJLevel` then](https://en.wikipedia.org/wiki/Term_symbol#Terms,_levels,_and_states)?).

### CODE REVIEW

File: docs/src/terms.md [hunk 1]
```diff
@@ -21,40 +19,80 @@ The [`Term`](@ref) objects can also be constructed with the [`@T_str`](@ref) str
...
 ```@docs
 IntermediateTerm
 intermediate_terms
-count_terms
+```
+
+### Disambiguating quantum numbers
+
+The [`IntermediateTerm`](@ref) type does not specify how to interpret the disambiguating
+quantum number ``ν``, or even what the type of it should be. In AtomicLevels, we use it two
```
Comment (jagot) [inline docs/src/terms.md:78]: we use ~~it~~ two

File: src/intermediate_terms.jl [hunk 2]
```diff
@@ -61,12 +61,45 @@ end
 
 # * Intermediate terms, seniority
 """
+    struct IntermediateTerm{T,S}
+
+Represents a term together with its extra disambiguating quantum number, labelled by `ν`.
```
Comment (jagot) [inline src/intermediate_terms.jl:66]: I would prefer to keep it as: "extra disambiguating quantum(s)" at least

File: src/intermediate_terms.jl [hunk 3]
```diff
@@ -61,12 +61,45 @@ end
...
+
+# Constructors
+
     IntermediateTerm(term, ν)
 
-Represents a `term` together with its extra disambiguating quantum
-numbers, labelled by `ν`, which should be sortable (i.e. comparable by
-`isless`). The most common implementation of this is a single quantum
-number, [`Seniority`](@ref).
+Constructs and intermediate term with term symbol `term` and disambiguating quantum number
```
Comment (jagot) [inline src/intermediate_terms.jl:82]: Same here: "quantum number(s)"

File: src/intermediate_terms.jl [hunk 4]
```diff
@@ -61,12 +61,45 @@ end
...
+`ν`.
+
+# Properties
+
+To access the term symbol and the disambiguating quantum number, you can use the
```
Comment (jagot) [inline src/intermediate_terms.jl:87]: And here

File: src/terms.jl [hunk 5]
```diff
@@ -3,9 +3,15 @@
 """
     struct Term
 
-Represent a term symbol ``{}^{2S+1}L_{J}`` with specific parity in LS-coupling. As determining
-valid ``J`` values is simple for given ``S`` and ``L`` (``|L - S| \\leq J \\leq L+S``), it
-is not specified.
+Represents a term symbol with specific parity in LS-coupling.
+
+Only the ``L``, ``S`` and parity values of the symbol ``{}^{2S+1}L_{J}`` are specified. To
+specify a _level_, the ``J`` value would have to be specified separately. The valid ``J``
```
Comment (jagot) [inline src/terms.jl:9]: Maybe then add a note that when used in a CSF, the level can be represented by `Level`?

Comment (jagot) [reply src/terms.jl:9]: Which of course is undocumented...

Comment (mortenpi) [reply src/terms.jl:9]: I might leave it out for now. Also `Level`, as currently implemented, relies on CSFs. We can maybe document it in #61.

**Comment by mortenpi** [comment]
I'd say this is good to go from my side.

**Pull request merged**